src/rust/bitbox02-rust/src/hww/api/cardano/sign_transaction/certificates.rs

```diff
@@ -88,6 +88,57 @@ pub async fn verify<'a>(
                 })
                 .await?;
             }
+            VoteDelegation(certificate::VoteDelegation {
+                keypath,
+                r#type,
+                drep_credhash,
+            }) => {
+                validate_address_shelley_stake(keypath, Some(bip44_account))?;
+                signing_keypaths.push(keypath);
+                let drep_type_name =  match certificate::vote_delegation::CardanoDRepType::try_from(*r#type)? {
+                   certificate::vote_delegation::CardanoDRepType::KeyHash => "Key Hash",
+                   certificate::vote_delegation::CardanoDRepType::ScriptHash => "Script Hash",
+                   certificate::vote_delegation::CardanoDRepType::AlwaysAbstain => "Always Abstain",
+                   certificate::vote_delegation::CardanoDRepType::AlwaysNoConfidence => "Always No Confidence",
+                };
+                confirm::confirm(&confirm::Params {
+                        title: params.name,
+                        body: &format!(
+                            "Delegate voting for account #{}?",
```

Comment:
I think UX would be a bit better if it was just a single line - it is already scrollable. Then the user does not have to worry about which type/drep it is, they can't see ahead.

If you think two screens are better anyway, it's better to not start it with "to ...", but maybe simply "Type: .... DRep: ...".

Reply:
I am split as I think 2 screens are a little bit more readable, but also a bit more confusing as I am not sure how many users will actually know what which type of delegation means so it might just unnecessarily confuse them. I made it 1 screen in the end.